kotest-assertions/kotest-assertions-shared/src/commonMain/kotlin/io/kotest/assertions/either.kt

```diff
@@ -0,0 +1,49 @@
+package io.kotest.assertions
+
+import io.kotest.common.ExperimentalKotest
+
+/**
+ * Runs multiple assertions and expects exactly one to succeed, will suppress all exceptions otherwise.
+ *
+ * ```
+ *   either {
+ *      "foo" shouldBe "bar"
+ *      "foo" shouldBe "foo
+ *      "foo" shouldBe "baz"
+ *   }
+ * ```
+ */
+@ExperimentalKotest
+suspend fun <T> either(assertions: suspend () -> T): T? {
```

Comment:
In hindsight this nomenclature might clash with arrow / functional programming.
What about calling this "one"

Reply:
I've updated this - let me know on slack if you disagree @jschneidereit